### ISSUE TITLE
Profile bio truncate behaviour

### ISSUE THREAD

**Issue by rohitkrai03** [openshiftio/openshift.io#3709]
> The bio code prevents users from typing more than 250 characters by removing the last X characters in the bio text s.t. it is 255 characters long.
> 
> This means if a user has something 255 characters long, and edits somewhere in the middle of his bio, the code will remove text at the end of the bio. This is definitely not a good thing to do.
> 
> I'd prefer the update to be disabled if a user goes over the limit and the bio section should show text, Bio cannot be over 255 characters or something similar so users are aware they need to modify their bio to be shorter.
> 
> Also curiously, new lines that a user enters in a bio are shown as spaces. This is definitely unexpected behavior from a user perspective.

![40659532-8d45fc3a-631c-11e8-986c-df8b143820ea](https://user-images.githubusercontent.com/6041994/40837942-ad4618b8-65b9-11e8-9b71-e587bcd84641.png)

![40659537-8f2b365a-631c-11e8-8594-a148c9bac42d](https://user-images.githubusercontent.com/6041994/40837976-d351c67e-65b9-11e8-935f-976c52f19d69.png)

Based on @jiekang's comment on a [PR](https://github.com/fabric8-ui/fabric8-ui/pull/2777). I think this issue needs help from the UX team.

cc: @alexeykazakov

**Comment by joshuawilson** [comment]
cc @openshiftio/uxd-team

**Comment by dlabrecq** [comment]
This particular functionality is based on Patternfly's "count remaining characters" pattern. The character length itself is something that was dictated by the back-end.

http://www.patternfly.org/pattern-library/widgets/#cnt-remaining-chars

The behavior for this particular page is to not allow the user to enter more than the character limit. However, the patternfly-ng directive does allow for that behavior to be changed. For example, we can allow the user to go beyond the character limit (by omitting the blockInputAtMaxLimit flag) and show a negative remaining count in red. You can also listen for the onOverCharsMaxLimit event and decide what should be disabled, if anything.

You can see that example below.
http://www.patternfly.org/patternfly-ng/#/remainingcharscount

Note that since this functionality is provided as an Angular directive, you provide the HTML text area element. If new lines are shown as spaces, you may need to change that behavior yourself? You may need to temporarily remove the directive and test if you obtain the same result?

**Comment by christianvogt** [comment]
Regardless of what implementation is currently there. Prevent adding more than 255 characters. Make sure pasting into the field also adheres to the 255 character limit. The component should never delete characters from the end of the text.

Keep newline handling as is for now even if they are collapsed to spaces.
Will wait on UXD to comment on handling new lines for display purposes and depending on answer we can open a new issue.

**Comment by serenamarie125** [comment]
Agreed that we shouldn't allow more than 255 characters.  IMO we should allow for new lines.
